Including documentation for radio element in interactive dialog (#3635)
mattermost/docs: source/developer/interactive-dialogs.rst

- **text**: Single-line plain text field. Use this for inputs such as names, email addresses or phone numbers.
- **textarea**: Multi-line plain text field. Use this field when the answer is expected to be longer than 150 characters. 
- **select**: Message menu. Use this for pre-selected choices. Can either be static menus or dynamic menus generated from users and public channels of the system. For more information on message menus, see :doc:`the documentation <interactive-messages>`.

> - **bool**: Checkbox option. Use this for binary selection.

Reviewer: Are you also interested in documenting this type?

Author: No, documentation about `bool` type had already been written.
https://docs.mattermost.com/developer/interactive-dialogs.html#checkbox-element

When writing about `radio type`, I noticed that `bool` is not here, so I just added it.

Reviewer: Missed that, thanks for the explanation :+1:

Checkbox Element
^^^^^^^^^^^^^^^^^^^^^^^

> In Mattermost version 5.16 and later, you can use checkbox elements. It looks like a plain text field with a checkbox to be selected. Below is an example of a checkbox element that asks for meeting feedback.

Reviewer 2: ```suggestion
From Mattermost 5.16 you can use checkbox elements. It looks like a plain text field with a checkbox to be selected. Below is an example of a checkbox element that asks for meeting feedback.
```

    "placeholder", "String", "(Optional) A string displayed to include a label besides the checkbox. Maximum 150 characters."
    

Radio Element
^^^^^^^^^^^^^^^^^^^^^^^

In Mattermost version 5.16 and later, you can use radio elements. It looks like a plain text field with a radio button to be selected. Below is an example of a radio element that asks for a department.

.. image:: ../../source/images/interactive-dialog-radio.png

.. code-block:: json

  {
    "display_name": "What department do you work in?",
    "name": "department",
    "type": "radio",
    "options": [
      {
        "text": "Engineering",
        "value": "engineering"
      },
      {
        "text": "Sales",
        "value": "sales"
      },
      {
        "text": "Administration",
        "value": "administration"

>       }

Reviewer: Given that the image above as a `default` element, it should also be added to this payload.

>     ]

Reviewer: Non blocking: I would also love to see the example above being extended with a `help_text` to ensure that every supported field is shown.

  }

The full list of supported fields are included below:

.. csv-table::
    :header: "Field", "Type", "Description"

    "display_name", "String", "Display name of the field shown to the user in the dialog. Maximum 24 characters."
    "name", "String", "Name of the field element used by the integration. Maximum 300 characters. You should use unique “name” fields in the same dialog."
    "type", "String", "Set this value to ``radio`` for a radio element."

>     "options", "String", "(Optional) An array of options for the radio element."

Reviewer: Correct, that is supposed to be an `Array`

    "placeholder", "String", "(Optional) A string displayed to include a label besides the checkbox. Maximum 150 characters."
    

Radio Element
^^^^^^^^^^^^^^^^^^^^^^^

> In Mattermost version 5.16 and later, you can use radio elements. It looks like a plain text field with a radio button to be selected. Below is an example of a radio element that asks for a department.

Reviewer 2: ```suggestion
From Mattermost 5.16 you can use radio elements. It looks like a plain text field with a radio button to be selected. Below is an example of a radio element that asks for a department.
```
